src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultCredentialRetriever.java

```diff
@@ -57,4 +65,76 @@ public static KeyVaultCredentials getCredentialById(String credentialID, Run<?,
+        }
+        SystemCredentialsProvider systemCredentialsProvider = SystemCredentialsProvider.getInstance();
+        List<AzureImdsCredentials> azureImdsCredentials =
+                DomainCredentials.getCredentials(systemCredentialsProvider.getDomainCredentialsMap(),
+                        AzureImdsCredentials.class,
+                        Collections.emptyList(),
+                        CredentialsMatchers.withId(credentialID));
+
+        if (!azureImdsCredentials.isEmpty()) {
+            return new AzureKeyVaultImdsCredential();
+        }
+
+        AzureKeyVaultCredential credential = null;
+        List<StandardUsernamePasswordCredentials> usernamePasswordCredentials =
+                DomainCredentials.getCredentials(systemCredentialsProvider.getDomainCredentialsMap(),
+                        StandardUsernamePasswordCredentials.class,
+                        Collections.emptyList(),
+                        CredentialsMatchers.withId(credentialID));
+        if (!usernamePasswordCredentials.isEmpty()) {
+            LOGGER.log(Level.FINE, format("Fetched %s as StandardUsernamePasswordCredentials", credentialID));
+            StandardUsernamePasswordCredentials usernamePasswordCredential = usernamePasswordCredentials.get(0);
+            credential = new AzureKeyVaultCredential(
+                    usernamePasswordCredential.getUsername(),
+                    usernamePasswordCredential.getPassword()
+            );
+        }
+
+        List<AzureCredentials> azureCredentials =
+                DomainCredentials.getCredentials(systemCredentialsProvider.getDomainCredentialsMap(),
+                        AzureCredentials.class,
+                        Collections.emptyList(),
+                        CredentialsMatchers.withId(credentialID));
+        if (!azureCredentials.isEmpty()) {
+            LOGGER.log(Level.FINE, format("Fetched %s as AzureCredentials", credentialID));
+            AzureCredentials azureCredential = azureCredentials.get(0);
+            credential = new AzureKeyVaultCredential(
+                    azureCredential.getClientId(),
+                    Secret.fromString(azureCredential.getPlainClientSecret())
+            );
+        }
+
+        if (credential == null) {
+            throw new AzureKeyVaultException(String.format("Credential: %s was not found for supported credentials " +
+                    "type.", credentialID));
+        }
+        if (!credential.isValid()) {
+            throw new AzureKeyVaultException("No valid credentials were found for accessing KeyVault");
+        }
+        return credential;
+    }
+
+    static SecretBundle getSecretBundle(KeyVaultClient client, AzureKeyVaultSecret secret, String keyVaultURL) {
+        try {
+            if (StringUtils.isEmpty(secret.getVersion())) {
+                return client.getSecret(keyVaultURL, secret.getName());
+            }
+            return client.getSecret(keyVaultURL, secret.getName(), secret.getVersion());
```

Comment:
Why this could work properly without setting version before this change :thinking:

Reply:
Possibly the azure-credentials plugin bump?

I verified on the previous release the behaviour as well...